UI/class.js

```diff
@@ -0,0 +1,656 @@
+                        return false;
+                    }
+                    break;
+                case 'photoLink':
+                    if (adItem.photoLink && typeof adItem.photoLink !== 'string'){
+                        return false;
+                    }
+                    break;
+                case 'hashTags':
+                    if (!adItem.hashTags || !adItem.hashTags.every(tag => typeof tag === 'string')){
+                        return false;
+                    }
+                    break;
+                case 'discount':
+                    if (!adItem.discount || typeof adItem.discount !== 'string'){
+                        return false;
+                    }
+                    break;
+                case 'validUntil':
+                    if (!adItem.validUntil || Object.prototype.toString.call(adItem.validUntil) !== '[object Date]') {
+                        return false;
+                    }
+                    break;
+                case 'rating':
+                    if (adItem.rating && (typeof adItem.rating !== 'number' || (adItem.rating < 0 && adItem.rating > 5))) {
+                        return false;
+                    }
+                    break;
+                case 'reviews':
+                    if (adItem.reviews && (typeof adItem.reviews.reviewText !== 'string' || typeof adItem.reviews.username !== 'string' || Object.prototype.toString.call(adItem.reviews.date) !== '[object Date]')) {
+                        return false;
+                    }
+                    break;
+                default:
+                    return false;
+            }
+            return true;
+        });
+    }
+
+    add(adItem) {
+        if (AdList.validate(adItem)) {
+            this._newAds.push(adItem);
+            return true;
+        }
+        return false;
+    }
+
+    edit(id, adItem) {
+        for (var param in adItem) {
+            if (param === 'id' || param === 'createdAt' || param === 'vendor') {
+                console.log('You can\'t change id, vendor or time');
+                return false;
+            }
+        }
+        if (!AdList.validate(adItem, Object.keys(adItem))) {
+            return false;
+        }
+
+        var editingAd = this.get(id);
+
+        for (var param in adItem) {
+            editingAd[param] = adItem[param];
+        }
+
+        return true;
+    }
+
+    remove(id) {
+        if (typeof id !== 'string') {
+            return false;
+        }
+        
+        var index = this._newAds.findIndex(adItem => adItem.id === id);
+        if (index === -1){
+              console.log('Incorrect id!');
+              return false;
+        }
+        
+        this._newAds.splice(index, 1);
+        return true;
+    }
+
+    addAll(ads) {
+        var invalidAds = [];
+
+        ads.forEach(adItem => {
+            if (!this.add(adItem)) invalidAds.push(adItem)
+        });
+
+        return invalidAds;
```

Comment:
I don't see where in this method you add valid items to `this._newAds` 🤔

Reply:
i add it in "this.add(adItem)"